ls1intum/Artemis
# `Programming Exercise`: Add solution and template participations to latest result endpoint

### Checklist
#### General

#### Server

### Motivation and Context


The `/programming-exercise-participations/{participationId}/latest-result-with-feedbacks` endpoint is used in the Themis app to retrieve student submissions in read-only mode (without performing assessment). The response coming from this endpoint did not contain solution and template participations. This prevented tutors from viewing solution and template repositories in read-only mode in Themis. This PR fixes the problem.

### Description

Added a new `@RequestParam withTemplateAndSolutionParticipations` to `getLatestResultWithFeedbacksForProgrammingExerciseParticipation` to fetch submission and template participations.

### Steps for Testing

Prerequisites:
- 1 Exercise with at least 1 student participation
- 1 Instructor

1. Log in as Instructor
2. Navigate to Course Management 
3. Find the exercise and click on the Scores button

> <img width="300" alt="Screenshot 2023-07-03 at 14 34 22" src="https://github.com/ls1intum/Artemis/assets/22798314/5260fec6-8071-4d33-91d9-ba560e5b7402">

4. Open one of the results in Online Editor
5. Make sure everything works as before on the Online Editor

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [ResultService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5097-5/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service/ResultService.html) | 91% | ✅ |
| [ProgrammingExerciseParticipationResource.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5097-5/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.web.rest/ProgrammingExerciseParticipationResource.html) | 86% | ✅ |

### Screencast

https://github.com/ls1intum/Artemis/assets/22798314/be658df4-cd6e-4253-b2f0-000c34d76fe7

Note: You do NOT need to test the Themis app. The video above just shows the effect of this PR on Themis.
